rust_src/src/frames.rs

```diff
@@ -87,6 +88,19 @@ impl LispFrameRef {
     }
 }
 
+cfg_if! {
+    if #[cfg(feature = "window-system-x11")] {
```

Comment:
Instead of an `Option` here you can use the rest of the enum values. `termcap` is the non-gui interface. `initial` is the first value. See frame.h and termhooks.h.

Reply:
Sorry for the long delay.
I have simplified it now to do a match on the frame's output method.
`initial` and `termcap` are not considered a GUI window, anything else is.
This way I don't even need this helper function. It should still be equivalent to the `FRAME_WINDOW_P` macro, assuming that builds for, say X  don't create `output_w32` frames.